src/System.Management.Automation/engine/Utils.cs

```diff
@@ -798,21 +798,119 @@ private static T GetPolicySettingFromConfigFile<T>(ConfigScope[] preferenceOrder
-            {nameof(ModuleLogging), @"Software\Policies\Microsoft\PowerShellCore\ModuleLogging"},
+            {nameof(ScriptExecution), @"Software\Policies\PowerShellCore"},
+            {nameof(ScriptBlockLogging), @"Software\Policies\PowerShellCore\ScriptBlockLogging"},
+            {nameof(ModuleLogging), @"Software\Policies\PowerShellCore\ModuleLogging"},
             {nameof(ProtectedEventLogging), @"Software\Policies\Microsoft\Windows\EventLog\ProtectedEventLogging"},
-            {nameof(Transcription), @"Software\Policies\Microsoft\PowerShellCore\Transcription"},
-            {nameof(UpdatableHelp), @"Software\Policies\Microsoft\PowerShellCore\UpdatableHelp"},
-            {nameof(ConsoleSessionConfiguration), @"Software\Policies\Microsoft\PowerShellCore\ConsoleSessionConfiguration"}
+            {nameof(Transcription), @"Software\Policies\PowerShellCore\Transcription"},
+            {nameof(UpdatableHelp), @"Software\Policies\PowerShellCore\UpdatableHelp"},
+            {nameof(ConsoleSessionConfiguration), @"Software\Policies\PowerShellCore\ConsoleSessionConfiguration"}
         };
 
+        private static readonly Dictionary<string, string> WindowsPowershellGroupPolicyKeys = new Dictionary<string, string>
+        {
+            {nameof(ScriptExecution), @"Software\Policies\Microsoft\Windows\PowerShell"},
+            {nameof(ScriptBlockLogging), @"Software\Policies\Microsoft\Windows\PowerShell\ScriptBlockLogging"},
+            {nameof(ModuleLogging), @"Software\Policies\Microsoft\Windows\PowerShell\ModuleLogging"},
+            {nameof(Transcription), @"Software\Policies\Microsoft\Windows\PowerShell\Transcription"},
+            {nameof(UpdatableHelp), @"Software\Policies\Microsoft\Windows\PowerShell\UpdatableHelp"},
+        };
+
+        private const string PolicySettingFallbackKey = "UseWindowsPowerShellPolicySetting";
+
         private static readonly ConcurrentDictionary<ConfigScope, ConcurrentDictionary<string, PolicyBase>> s_cachedPoliciesFromRegistry =
             new ConcurrentDictionary<ConfigScope, ConcurrentDictionary<string, PolicyBase>>();
 
         private static readonly Func<ConfigScope, ConcurrentDictionary<string, PolicyBase>> s_subCacheCreationDelegate =
             key => new ConcurrentDictionary<string, PolicyBase>(StringComparer.OrdinalIgnoreCase);
 
+        /// <summary>
+        /// Read policy settings from a registry key into a policy object.
+        /// </summary>
+        /// <param name="instance">Policy object that will be filled with values from registry.</param>
+        /// <param name="instanceType">Type of policy object used.</param>
+        /// <param name="gpoKey">Registry key that has policy settings.</param>
+        /// <returns>True if any property was successfully set on the policy object.</returns>
+        private static bool ReadPolicySettingsFromRegistryKey(object instance, Type instanceType, RegistryKey gpoKey)
```

Comment:
Should it be `TrySetPolicySettingsFromRegistryKey`?

Reply:
Renamed.